File: 0053-maximum-subarray/0053-maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        max_sum_subarray = float('-inf')
        current_sum = 0

        for num in nums:
          # Kadane Algo : add → update maximum → reset if -ve
          current_sum = max(current_sum + num, num)
          max_sum_subarray = max(current_sum, max_sum_subarray)
        
        return max_sum_subarray

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE;
        int currentSum = 0;

        for (int num : nums) {
          // Kadane Algo : add → update maximum → reset if -ve
          currentSum = Math.max(currentSum + num, num);
          maxSum = Math.max(currentSum, maxSum);
        }

        return maxSum;
    }
}